gen/groundobjectsgen.py

```diff
@@ -259,6 +262,56 @@ def generate(self) -> None:
         self.generate_static(Fortification.Workshop_A)
 
 
+class SceneryGenerator(BuildingSiteGenerator):
+    def generate(self) -> None:
+        if isinstance(self.ground_object, SceneryGroundObject):
+            trigger_zone = self.generate_trigger_zone(self.ground_object)
```

Comment:
Don't need (or want) to do this if it's dead, right?

Reply:
It is necessary to generate new Triggers (which have TriggerZones in them) on a mission per mission basis as the Triggers are what control if the building is displayed dead or not.

Reply:
Better answer, TriggerZones are necessary to make a DestructionEvent trigger.